pkg/registry/servicecatalog/binding/strategy.go

```diff
@@ -120,8 +121,23 @@ func (bindingRESTStrategy) PrepareForUpdate(ctx genericapirequest.Context, new,
 	if !ok {
 		glog.Fatal("received a non-binding object to update from")
 	}
-	newServiceInstanceCredential.Spec = oldServiceInstanceCredential.Spec
 	newServiceInstanceCredential.Status = oldServiceInstanceCredential.Status
+
+	// TODO: We currently don't handle any changes to the spec in the
+	// reconciler. Once we do that, this check needs to be removed and
+	// proper validation of allowed changes needs to be implemented in
+	// ValidateUpdate. Also, the check for whether the generation needs
+	// to be updated needs to be un-commented.
+	newServiceInstanceCredential.Spec = oldServiceInstanceCredential.Spec
```

Comment:
Can we throw some error instead of silently ignoring any changes to the spec?
This behavior can be confusing for users, even though it's temporary.

Reply:
See my comment here: https://github.com/kubernetes-incubator/service-catalog/pull/1151#issuecomment-326170780

Reply:
Looks like we can return errors in the `ValidateUpdate` method though (below).
```go
	// ValidateUpdate is invoked after default fields in the object have been
	// filled in before the object is persisted.  This method should not mutate
	// the object.
	ValidateUpdate(ctx genericapirequest.Context, obj, old runtime.Object) field.ErrorList
```
Both new and old objects are passed there, so we can compare spec and return error if they are not identical.

Reply:
Elsewhere in Kubernetes, we ignore changes to fields that cannot be changed.

Reply:
See #1182

Reply:
Fair enough